### ISSUE TITLE
TODO: Add outloader (test ood data loader)

### ISSUE THREAD

**Issue by elainyilanchen** [ql2501/Capstone-ood#12]
1. √Download ood dataset (e.g. texture) to DATA folder
2. √For each dataset, add dataset class to `CoOp_works\CoOp\datasets\{dataset name}.py`:

 - √use dummy label (not used).
 - √**Not implemented, use whole dataset as test set**  *may split to val and test set, depending on ID dataset val and test settings* **(data size concern)**
 - √add one dummy "train" image with dummy label: to avoid error when loading super class `DatasetBase`

2.5 √Add `texture.yaml`

3. Add cfg for OOD dataset name

4. In `CoOp_works\Dassl.pytorch\dassl\data\datasets\build.py`: modify to optional read OOD dataset name from cfg
5. In `CoOp_works\Dassl.pytorch\dassl\data\data_manager.py`：

 - line 62, add loading `dataset_ood`
 - line 138, add `out_loader` (copy paste test loader and modify) (**using same cfg of sampler, batch size, transformation as test**
 - modify `show_dataset_summary` by adding `self.dataset_ood` in init to show info. **Not sure if this will increase space usage**
